### PR TITLE
Increase XP tracker accuracy to 2 decimal places

### PR DESCRIPTION
**Pull request by Ericwh2827**
Increased XP accuracy in tracker to 2 decimal places.

increase xp accuracy

### CODE REVIEW

**Comment by deathbeam** [comment]
Why should there be toggle? That seems silly

**Comment by deathbeam** [comment]
I would do this entire change like this: Remove the toggle, make the getXpProgress returns decimals with increased accuracy, display that instead. Would make the change like 2 lines instead of 60

**Comment by Ericwh2827** [comment]
> I would do this entire change like this: Remove the toggle, make the getXpProgress returns decimals with increased accuracy, display that instead. Would make the change like 2 lines instead of 60

I have updated my pull request to remove the toggle so that it always show XP to 2 decimal places.

File: runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpInfoBox.java [hunk 1]
```diff
@@ -197,8 +198,8 @@ private void rebuildAsync(boolean updated, boolean skillPaused, XpSnapshotSingle
 			actionsLeft.setText(htmlLabel("Actions: ", xpSnapshotSingle.getActionsRemainingToGoal()));
 
 			// Update progress bar
-			progressBar.setValue(xpSnapshotSingle.getSkillProgressToGoal());
-			progressBar.setCenterLabel(xpSnapshotSingle.getSkillProgressToGoal() + "%");
+			progressBar.setValue((int) xpSnapshotSingle.getSkillProgressToGoal());
+			progressBar.setCenterLabel(new DecimalFormat("0.00").format(xpSnapshotSingle.getSkillProgressToGoal()) + "%");
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpInfoBox.java]: you should probably use StackFormatter class to format this, or declare decimal format as constant at top

Comment (Ericwh2827) [reply runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpInfoBox.java]: I opted to declare the decimal format as a constant at the top.

File: runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpInfoBox.java [hunk 2]
```diff
@@ -87,6 +88,8 @@
 
 	private boolean paused = false;
 
+	private static final DecimalFormat twoDecimalPlaces = new DecimalFormat("0.00");
```
Comment (Kamielvf) [inline runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpInfoBox.java]: should be named UPPER_CASE something like `TWO_DECIMAL_FORMAT`

Comment (Ericwh2827) [reply runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpInfoBox.java]: Whoops. Fixed.

File: runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpInfoBox.java [hunk 3]
```diff
@@ -87,6 +88,8 @@
 
 	private boolean paused = false;
 
+	private static final DecimalFormat TWO_DECIMAL_FORMAT = new DecimalFormat("0.00");
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpInfoBox.java]: consts go at top of the file, before instance fields

**Pull request merged**